File: Obj Oriented/practice.lua
--[[ -- Runtime
cd "c:\Users\stefa\codepractice"
lua "Obj Oriented\practice.lua"
]]-- This is a multi-line comment. The code between the brackets is not executed.

print("Hello, World!") -- This is a comment. This line prints "Hello, World!" to the console

var = 10 -- This line assigns the value 10 to the variable 'var'
print(var) -- This line prints the value of 'var' to the console

for i = 1, 5 do -- This line starts a numeric loop that will iterate from 1 to 5. i is a placeholder variable that takes on the value of each iteration
    print("Iteration: " .. i) -- This line prints the current iteration number
end -- This marks the end of a block of code, such as a loop or function

list = {1, 2, 3, 4, 5} -- This line creates a table (similar to an array) with the values 1 through 5
for i, v in ipairs(list) do -- This line starts a generic loop that iterates over the elements of a table
    print("Index: " .. i .. ", Value: " .. v) -- This line prints the index and value of each element in the table. v is a placeholder variable that takes on the value of each element in the table during the iteration
end -- This marks the end of the loop block

function printValues(number)
    tonumber(number) -- This line converts the input 'number' to a numeric type, if it is not already
    for i = 0, number do -- This line starts a numeric loop that will iterate from 0 to the value of 'number'
        print(i) -- This line prints the current value of i during each iteration
    end -- This marks the end of the loop block
end -- This marks the end of the function block

printValues(4) -- This line calls the function 'printValues' with the argument 4, which will print the numbers from 0 to 4 to the console

io.write("Enter a number: ") -- This line prompts the user to enter a number
input = io.read() -- This line reads the user's input and stores it in the variable 'input'
num = {0, 1, 2, 3, 4, 5} -- This line creates a table with the values 0 through 5
for i, v in ipairs(num) do -- This line starts a generic loop that iterates over the elements of the 'num' table
    print("The console chose: " .. v) -- This line prints the current value of v during each iteration
    print("Index: " .. i .. ", Value: " .. v) -- This line prints the index and value of each element in the 'num' table during the iteration
    break -- This forcefully exits the loop after the first iteration, so only the first element of the 'num' table will be printed
end

if input == num[0] or input == num[1] or input == num[2] or input == num[3] or input == num[4] then -- This line checks if the user's input is equal to any of the elements in the 'num' table
    print("You guessed correctly the number! You said: " .. input) -- If the condition is true, this line prints "You win!" to the console
else -- This line starts an alternative block of code that will execute if the previous condition was false
    print("You guessed an incorrect number or gave nil! You said: " .. input) -- If the user's input does not match any of the elements in the 'num' table, this line prints "You lose!" to the console
end -- This marks the end of the if-else block

File: Obj Oriented/practice.py
""" # Runtime
python "Obj Oriented\pratice.py"
"""

# This is a variable that stores a number and prints it to the console
num = input("Enter a number: ") # num is the variable. input() is the user input to be accepted
num = int(num) # int() converts the input to an integer
print("The number you entered is: " + str(num)) # This prints the variable. str() converts the number back to a string for concatenation with a + sign

for i in range(5): # This is a loop that will run 5 times. i is the variable that will take on the values from 0 to 5
    print("This is iteration number: " + str(i)) # This prints the current iteration number. str() converts the integer to a string for concatenation

def add_numbers(a, b): # This is a function that takes two parameters, a and b
    a = input("Enter the first number, a: ") # This prompts the user to enter the first number and stores it in variable a
    b = input("Enter the second number, b: ") # This prompts the user to enter the second number and stores it in variable b
    a = int(a) # Converts the input to an integer
    b = int(b) # Converts the input to an integer
    print("The sum of the two numbers is: " + str(a + b)) # This prints the sum of a and b

add_numbers(0, 0) # This calls the function add_numbers with dummy arguments since the function will prompt for user input

if num == int(input("Enter a number to compare with: ")): # This checks if the variable num is equal to the user input
    print("The numbers are equal!") # This prints if the numbers are equal
elif num != int(input("Enter a number to compare with: ")): # This checks if the variable num is not equal to the user input
    print("The numbers are not equal!") # This prints if the numbers are not equal
else: # This is the else statement that will execute if none of the above conditions are met
    print("Invalid input!") # This prints if the input is invalid

class variables: # creates a class with the name variables
    def __init__(self, name, value): # This is the constructor method that initializes the class with self, name, and value parameters
        "The __init__ method is called when an instance of the class is created. It initializes the instance variables with the provided values. Self is a reference to the current instance of the class, and it allows you to access the instance variables and methods within the class. Name and value are parameters that are passed when creating an instance of the class, and they are used to set the instance variables self.name and self.value."
        self.name = name # This assigns the name parameter to the instance variable self.name
        self.value = value # This assigns the value parameter to the instance variable self.value